### PR TITLE
[OCPCRT-169] Do not show i icon if no inconsistency exist

### PR DESCRIPTION
**Pull request by ashwindasr**
Follows https://github.com/openshift/release-controller/pull/575

Currently, we display the `i` icon next to all nightlies if the latest one has inconsistency. With this update, it will not display that icon.

### CODE REVIEW

File: cmd/release-controller-api/http.go
Comment (hoxhaeris) [inline cmd/release-controller-api/http.go:1500]: The `doesInconsistencyExist` method is called for every stream. That might not be ideal (it will subsequently call `getStreamFromTag`, which lists the imagestreams every time).
Can you perhaps fetch the imagestreams once in the `httpReleases` release method, and use that data to render the page, rather than calling `c.releaseLister.List(labels.Everything()` for every stream?

**Review by hoxhaeris — CHANGES_REQUESTED** [review]

File: cmd/release-controller-api/http.go
Comment (ashwindasr) [reply cmd/release-controller-api/http.go:1500]: In python, `strings.Contains(tag.Name, "nightly")` is `False` then the check after `&&` is not executed. Is it the same for Go?

Comment (ashwindasr) [reply cmd/release-controller-api/http.go:1500]: I think its called short circuiting.

Not sure if we are talking about different things now haha?

Comment (hoxhaeris) [reply cmd/release-controller-api/http.go:1500]: Afaik, the same applies to golang, if the first statement is evaluated as false, the second one will not execute. The `doesInconsistencyExist` method will be called for every nightly stream (too many times).  You don't need to `c.releaseLister.List(labels.Everything()` separately for every nightly, but rather do it once and use the data subsequently.

Comment (hoxhaeris) [reply cmd/release-controller-api/http.go:1500]: Afaik, the same applies to golang, if the first statement is evaluated as false, the second one will not execute. The `doesInconsistencyExist` method will be called for every nightly stream (too many times).  You don't need to `c.releaseLister.List(labels.Everything()` separately for every nightly, but rather do it once and use the data subsequently.

Comment (ashwindasr) [reply cmd/release-controller-api/http.go:1500]: Ahh! Makes sense. Will work on the change.

Comment (ashwindasr) [reply cmd/release-controller-api/http.go:1494]: Not sure if this is what you envisioned https://github.com/openshift/release-controller/pull/577/commits/243fe90b2e3b25003091ca657cdd55dc31b0ac4d

**Comment by hoxhaeris** [comment]
/hold
@ashwindasr, I will tag the PR. Feel free to unhold when you're ready to merge it.

**Review by hoxhaeris — APPROVED** [review]

**Comment by hoxhaeris** [comment]
I am not on the approvals list actually, I can't add the `approve` label. @bradmwilliams, when possible, can you please have a look as well?

**Comment by bradmwilliams** [comment]
@hoxhaeris create a PR to add yourself as an approver.

**Pull request merged**
